unit_tests/test_tlslite_ocsp.py

```diff
@@ -0,0 +1,151 @@
+"k82y7fOo9N96hDrWZLF7h+p2Bg8og+eUD9kNNGPx2/iLfNYYUDynU4Ay3wOxRzb8MHTw7OpevlEN"
+"GNUvcGYyOSWwMQHk8P0BCJZA8RpXuEkW/ep0ZSnNVMpbjtSpSE1i0OxMdKbfJTzm+l87FMXz/8oX"
+"p+YNJ+UJfB6ALag="))
+resp_malformed = a2b_base64("MAMKAQE=")
+resp_internal = a2b_base64("MAMKAQI=")
+resp_trylater = a2b_base64("MAMKAQM=")
+resp_sigreq = a2b_base64("MAMKAQU=")
+resp_unauthorized = a2b_base64("MAMKAQY=")
+resp_nonext = a2b_base64(str(
+"MIIBsAoBAKCCAakwggGlBgkrBgEFBQcwAQEEggGWMIIBkjB8ohYEFPW6BZQHpc7jISl5Z3Tq3X0p"
+"dI9AGA8yMDE4MDExNzEzMzgyOVowUTBPMDowCQYFKw4DAhoFAAQUkBUkEUsQTCuZ8rp/mrdlULbf"
+"yK4EFPW6BZQHpc7jISl5Z3Tq3X0pdI9AAgEBgAAYDzIwMTgwMTE3MTMzODI5WjANBgkqhkiG9w0B"
+"AQUFAAOCAQEAgX46PdN6W6gLbOuWNwTHAwJ79agqjADHLtvYyISVCenKmaCMrhoe6jcutw1A4wz4"
+"9BKfYqnO9AL1LYHOJ6ZPlzJx/nEiFue8HXb+ChiwG+nEhXprug+wP/APuKSOaKH2kcQf4Jtuv9cz"
+"n/2PCaVmC+ErEThuGTZouT4eEIFMUGGDH+nZFHbl+DNm6R3+D7atOE1gDBO2LDJqIoZvaZTSpY+4"
+"djZaiTdWAdOcUnnBhlhBvjg8nv6zxJ9ERBqm5P9cffnYVAJeGqylq/WFT/Ni6MprXgYoZq9bc7tk"
+"PGGh5CnrOhcIQCUIX+ceM6ruxdraiPJALpY1gZz7SY53GQsXfg=="))
+
+class TestOCSP(unittest.TestCase):
+    def test___init__(self):
+        resp = OCSPResponse(resp_OK)
+        self.assertEqual(OCSPRespStatus.successful, resp.responseStatus)
+    
+    def test_malformedrequest(self):
+        resp = OCSPResponse(resp_malformed)
+        self.assertEqual(OCSPRespStatus.malformedRequest, resp.responseStatus)
+    
+    def test_internalerror(self):
+        resp = OCSPResponse(resp_internal)
+        self.assertEqual(OCSPRespStatus.internalError, resp.responseStatus)
+
+    def test_trylater(self):
+        resp = OCSPResponse(resp_trylater)
+        self.assertEqual(OCSPRespStatus.tryLater, resp.responseStatus)
+
+    def test_sigrequired(self):
+        resp = OCSPResponse(resp_sigreq)
+        self.assertEqual(OCSPRespStatus.sigRequired, resp.responseStatus)
+
+    def test_unauthorized(self):
+        resp = OCSPResponse(resp_unauthorized)
+        self.assertEqual(OCSPRespStatus.unauthorized, resp.responseStatus)
+
+    def test_type_id_pkix_ocsp_basic(self):
+        resp = OCSPResponse(resp_OK)
+        self.assertEqual(bytearray([43, 6, 1, 5, 5, 7, 48, 1, 1]), resp.responseType)
+
+    def test_responderID(self):
+        resp = OCSPResponse(resp_OK)
+        self.assertEqual(bytearray([4, 20, 156, 77, 0, 153, 0, 14, 139, 176, 1, 129, 
+                                    117, 161, 186, 240, 208, 37, 215, 160, 28, 71]), 
+                                    resp.responderID)
+    
+    def test_producedAt(self):
+        resp = OCSPResponse(resp_OK)
+        self.assertEqual(bytearray(b"20171113135112Z"), resp.producedAt)
+
+    def test_signatureAlgorithm(self):
+        resp = OCSPResponse(resp_OK)
+        self.assertEqual(bytearray([42, 134, 72, 134, 247, 13, 1, 1, 11]), resp.signatureAlgorithm)
+
+    def test_signature(self):
+        resp = OCSPResponse(resp_OK)
+        self.assertEqual(bytearray([0, 155, 245, 236, 104, 50, 168, 180, 30, 1, 17, 165,
```

Comment:
bad indentation, if after the parentheses there is code/parameters, following lines should be on the same level of indent as the first line, like this:
```
a = str("some text"
        " continues")
```

but if the code is too long to fit comfortably on one or two lines, it's better to start it on next line with 4-space indent:
```
a = str(
    "some text continues for many many paragraphs"
    " and almost never ends")
```